src/Devtools/Files.hs

```diff
@@ -0,0 +1,44 @@
+module Devtools.Files
+  ( findHaskellFiles
+  )
+where
+
+import           Devtools.Prelude
+import           System.Path                    ( (</>) )
+
+import qualified Data.List                     as List
+import qualified System.Path                   as Path
+import qualified System.Path.Directory         as Path
+
+findHaskellFiles :: [Path.RelDir] -> IO [Path.RelFile]
+findHaskellFiles = fmap List.concat . traverse findFilesRecursive
+
+findFilesRecursive :: Path.RelDir -> IO [Path.RelFile]
+findFilesRecursive path = do
+  filePath <- maybe (fail "File doesn't not exist") pure $ Path.fileFromDir path
+
+  Path.doesFileExist filePath >>= \case
+    True -> pure [filePath]
+    _    -> Path.doesDirectoryExist path >>= \case
+      True -> findFilesRecursive' path
+        <&> List.filter ((== ".hs") . Path.takeExtension)
```

Comment:
As a followup we may want to include `hs-boot` files here.

Reply:
Yes, going to.